### ISSUE TITLE
UEFI Secure Boot?

### ISSUE THREAD

**Issue by ghost** [Byzantium/Byzantium#206]
Hi

Are you guys aware of this new issue? I found a motherboard that had the option:

Asus P8H61-MX R2.0, UEFI BIOS 0904 x64

Had in the boot section, secure boot with 2 options, Other OS or UEFI Windows.
1  Other OS is just normal.
2  UEFI Windows gives new management options to store a PK cert from a UEFI formatted USB stick.

I tried Byzantium on both options and works fine, but i read that UEFI wasn't active so it didn't care i was boot Linux. I had no "PK" stored.

Anyway, my point is, this could effect the project in the future, and all Linux.

Later

**Comment by virtadpt** [comment]
I've been both waiting for and dreading a ticket like this.

That Byzantium boots in 'Other OS' mode is reassuring.  As for UEFI as it's meant to be used, I have no stonking idea.  I know that Redhat has released an ultra-tiny, signed bootloader-shim that is UEFI compatible, but I don't know how that would plug into a live distribution.

Would you be interested in helping us figure this out?  I don't have any UEFI gear, and I don't know if anybody else does, either.

**Comment by ghost** [comment]
"I've been both waiting for and dreading a ticket like this". Yeah, i understand. I thought it was better it was out in the open. I hate it too. Microsoft is trying to get rid of Linux.

This motherboard is not mine, but i have access to it from time to time. I put the model here so you all can find & check out the manual if you want.

Solutions:
1) Would need every motherboard, forever to have the same options in their UEFI BIOS, Other OS & UEFI Windows. Add to the instructions to Byzantium on how to switch over to Other OS mode before booting. Harder for end users. My preferred option.

2) Byzantium to join a few other Linux versions to create their own custom PK cert and get it into some motherboards, so you could boot using secure boot. You would be dependent on the motherboards you choose, and this would be Damm hard to pull off, but easier for the end users. 

3) Submit to Microsoft and add the Microsoft PK cert into Byzantium for secure booting. Easy for end users. I think this is how Red Hat is doing it.

4) Hack every motherboards UEFI, and version and extract the key out of the PK cert. Then you can put the key into Byzantium. This would be impossible, not the cracking, but every motherboard and version from every brand! I think some guy from Germany did the same idea for the PS3, and i heard he ended up in jail, thanks to $ony!

Would you be interested in helping us figure this out? Well, I don't know how i can help, I'm not very good with Linux. I do have a little time available.

**Comment by ghost** [comment]
You could also boot by virtual box is another way from Windows 8 with USB wifi. Skipping the whole UEFI thing.

Anyway, could you consider putting this in if you choose option 1, for the instructions:

If you have a modern PC with UEFI, reboot your PC/Laptop, and press the usual Del,F2, etc on the bottom of the screen, a new screen will appear, using your mouse in the UEFI BIOS, choose Advance option, then click on "Boot"  scroll down and click on secure booting, then select Other OS, then back, and save reboot (usually F10)
Now booting will work as normal. Please write this down if you are new to this, as if you make a mistake, you can reverse your actions.

That may help new users.

**Comment by jbshirk** [comment]
This is an interesting, insightful and probably important article
related to this subject:
http://boingboing.net/2012/08/23/civilwar.html

**Comment by virtadpt** [comment]
That's a good idea, willi.

**Comment by virtadpt** [comment]
Hrm....: https://wiki.archlinux.org/index.php/Unified_Extensible_Firmware_Interface

**Comment by virtadpt** [comment]
So, here's how I had to do it at work on some of our Sony laptops:
- Boot into the system configuration, which is highly system-dependent.
- Enable BIOS emulation.
- Turn off all of the SafeBoot crap scattered through the system configuration screens.
- Change the boot order so that USB devices were tried first, the optical drives second.
- Enabled AHCI mode (this tripped me up a half-dozen times in a row).
- Figured out which button I had to press during boot to enter the boot device selection screen because UEFI was ignoring the explicitly set boot device list (on our Vaios it was a little "ASSISTANCE" button above the keyboard.
